### ISSUE TITLE
ERROR NullInjectorError: R3InjectorError(Or)[t -> t -> t]:    NullInjectorError: No provider for t!

### ISSUE THREAD

**Issue by tripurarisingh1994** [dpa99c/cordova-launch-review#29]
Hi, I am getting the issue ERROR NullInjectorError: R3InjectorError(Or)[t -> t -> t]: 
  NullInjectorError: No provider for t! - on the **Android**

Thanks

**Comment by dpa99c** [comment]
I don't believe that error is related to this plugin - please build and run the [example project](https://github.com/dpa99c/cordova-launch-review-example) to verify your build environment.

**Comment by pablojjimenez** [comment]
I fixed it:
Importing `LaunchReview` on the providers section from `import { LaunchReview } from '@awesome-cordova-plugins/launch-review/ngx';`

**Comment by BigBalli** [comment]
This is an Angular dependency injection error, not related to the cordova-launch-review plugin itself. The `NullInjectorError: No provider for t!` message means Angular can't find a required provider in your module.

If you're using the Ionic/Angular wrapper (`@awesome-cordova-plugins/launch-review`), make sure you've added `LaunchReview` to the `providers` array in your `app.module.ts`:

```typescript
import { LaunchReview } from '@awesome-cordova-plugins/launch-review/ngx';

@NgModule({
  providers: [
    LaunchReview,
    // ...
  ]
})
```

Also confirm you're importing from the `/ngx` path (not the root), which is required for Angular's DI to work properly.
